src/fabric/src/fabric_doc_update.erl

```diff
@@ -39,10 +39,15 @@ go(DbName, AllDocs0, Opts) ->
     Acc0 = {length(Workers), length(AllDocs), list_to_integer(W), GroupedDocs, dict:new()},
     Timeout = fabric_util:request_timeout(),
     try rexi_utils:recv(Workers, #shard.ref, fun handle_message/3, Acc0, infinity, Timeout) of
+        {ok, {_, []}} ->
+            erlang:error({internal_server_error, "No Responses For Document Update"});
```

Comment:
hmmm good point, I didn't consider that. What if we added:

```
{ok, {_, []}} when length(AllDocs) > 0
```
?

Reply:
I think if we go with the idea that we always expect the number of response to be the number of docs in the request, otherwise it's an error then we might not even need to handle this special case at all?

Reply:
I was thinking about this for a bit. Running your concurrency tests, we never see this `{error, internal_server_error}` get added to the list of responses. It's always just an `[]` `force_reply` is never called in that case, so we still might need this special case.

Reply:
nvm, I think I have a cleaner solution